doc/whats-new.rst

```diff
@@ -7225,6 +7225,7 @@ Breaking changes
   such `'DJF'`:
 
   .. ipython:: python
+      :okwarning:
```

Comment:
Can we change the string rather than allow warnings? Because we'll likely raise an error at some point... (and at the other location from the previous PR if possible)

Reply:
Yes I agree with that.

I saw how Pandas did it in their docs where 2.2 broke an older 'whats new' file, so will replicate what they did.